openmmtools/states.py

```diff
@@ -457,17 +458,22 @@ def pressure(self):
         If it is set to None, the barostat will be removed.
 
         """
-        barostat = self._barostat
-        if barostat is None:
-            return None
-        return barostat.getDefaultPressure()
+        return self._pressure
 
     @pressure.setter
-    def pressure(self, value):
-        # Invalidate cache if the ensemble changes.
-        if (value is None) != (self._barostat is None):
-            self._cached_standard_system_hash = None
-        self._set_system_pressure(self._system, value)
+    def pressure(self, new_pressure):
```

Comment:
Would it be easier and cleaner to instead store the "old state" as a complete object? Or do you think storing each component is safer?